frontend/src/modules/Vfp/utils/VfpDataAccessor.ts

```diff
@@ -95,35 +109,59 @@ export class VfpDataAccessor {
     }
 
     getWfrType(): WFR_api {
-        return this._vfpTable.wfr_type
+        if ("wfr_type" in this._vfpTable) { // This means that it is a VFPPROD table
+            return this._vfpTable.wfr_type
+        }
+        throw Error("WFR type is not valid for VFPINJ tables.")
     }
 
     getGfrType(): GFR_api {
-        return this._vfpTable.gfr_type
+        if ("gfr_type" in this._vfpTable) { // This means that it is a VFPPROD table
+            return this._vfpTable.gfr_type
+        }
+        throw Error("GFR type is not valid for VFPINJ tables.")
     }
 
     getAlqType(): ALQ_api {
-        return this._vfpTable.alq_type
+        if ("alq_type" in this._vfpTable) { // This means that it is a VFPPROD table
+            return this._vfpTable.alq_type
+        }
+        throw Error("ALQ type is not valid for VFPINJ tables.")
     }
 
-    getBhpValues(thpIndex: number, wfrIndex: number, gfrIndex: number, alqIndex: number) : number[] {
-        const nbWfrValues = this._vfpTable.wfr_values.length
-        const nbGfrValues = this._vfpTable.gfr_values.length
-        const nbAlqValues = this._vfpTable.alq_values.length
-        const nbFlowRates = this._vfpTable.flow_rate_values.length
-        const startIndex = nbFlowRates*(nbAlqValues*(nbGfrValues*(nbWfrValues*thpIndex+wfrIndex)+gfrIndex)+alqIndex)
-        return this._vfpTable.bhp_values.slice(startIndex, startIndex+nbFlowRates)
+    getVfpProdBhpValues(thpIndex: number, wfrIndex: number, gfrIndex: number, alqIndex: number) : number[] {
+        if ("wfr_values" in this._vfpTable) { // This means that it is a VFPPROD table
```

Comment:
Consider using suggested `isProdTable`-utlity for if-check.

I'd reverse the if-check to, s.t. error is thrown in if-statement and larger amount of code is outside if:

```
if (!isProdTable(this._vfpTable)){
  throw Error("The getVfpProdBhpValues function is not valid for VFPINJ tables.")
}

const nbWfrValues = this._vfpTable.wfr_values.length
const nbGfrValues = this._vfpTable.gfr_values.length
const nbAlqValues = this._vfpTable.alq_values.length
const nbFlowRates = this._vfpTable.flow_rate_values.length
const startIndex = nbFlowRates*(nbAlqValues*(nbGfrValues(nbWfrValues*thpIndex+wfrIndex)+gfrIndex)+alqIndex)
return this._vfpTable.bhp_values.slice(startIndex, startIndex+nbFlowRates)